### DISCUSSION TITLE
Render math symbols in functions

### DISCUSSION THREAD

**Discussion by jassler** [typst/typst#1046] [Q&A] [answered]
I wanted to create a shortcut for writing total orders, i.e., 'a succ.tilde b succ.tile c' (should come out something like "a ≥ b ≥ c"). I managed to create a simple join function that does this:

```typst
#let pr(els) = els.join(" succ.tilde ")
We have the power relation $#pr(("a", "b", "c", "d"))$.
```

Unfortunately, it doesn't seem to enter the [succ.tilde](https://www.tutorialspoint.com/tex_commands/succsim.htm) symbol, it just renders normal text.

<img width="549" alt="rendered pdf" src="https://user-images.githubusercontent.com/24298790/235359626-56acfc29-d9b5-493e-a05e-e2fd399a2131.png">

I tried adding dollar signs all around the place, to no avail. I also couldn't really find something in the documentation on how use functions inside of math environments.

How do I get functions that return a string to render math symbols?

**Comment by EpicEricEE** [comment, marked as answer]
As you are using quotes, you are joining your list with the text "succ.tilde", and not with the symbol. You can use dollar signs in the join instead:
```
#let pr(els) = els.join($succ.tilde$)
We have the power relation $#pr(("a", "b", "c", "d"))$.
```
![image](https://user-images.githubusercontent.com/7191192/235361128-7a618c6c-3936-46fc-94d5-f938cc4fe08b.png)

**Reply by jassler** [reply]
Does this also work with the replace method? i.e.,

```
#let pr(str) = str.replace(">", $succ.tilde$)
We have the power relation $#pr("a > b > c > d")$
```

**Reply by EpicEricEE** [reply]
Not like that, because `replace` expects a string, but `$succ.tilde` is of type `Content`. Maybe there is another way, but I'm not familiar with one.

**Comment by jassler** [comment]
Aah nice thanks! I'm not used to the fact that I don't have to use quotes for everything text related.

On that note, I tried doing the same with [two dots `..`](https://typst.app/docs/reference/scripting#bindings), thinking that I could save on extra braces that way.

```
#let pr(..els) = els.join( $succ.tilde$ )
We have the power relation $#pr("a", "b", "c", "d")$.
```

But this gives me the error `type arguments has no method 'join'`. Are those dots not treated as arrays?

**Reply by N4tus** [reply]
`..els` does not make it an array but an argument. It collects positional and keyword arguments. In your case you can get the positional arguments with the `pos()` method:
```
#let pr(..els) = els.pos().join( $succ.tilde$ )
We have the power relation $#pr("a", "b", "c", "d")$.
```
